### PR TITLE
[CK_TILE] wa prec, remove sgpr offset for inline asm

### PR DESCRIPTION
**Pull request by carlushuang**
1. WA precision issue in rocm-6.2 (compute-artifactory.amd.com:5000/rocm-plus-docker/framework/compute-rocm-dkms-no-npi-hipclang:14119_ubuntu22.04_py3.10_pytorch_release-2.3_db8da3b)
2. remove sgpr offset to speed up issue

### CODE REVIEW

**Review by poyenc — DISMISSED** [review]

**Review by poyenc — DISMISSED** [review]

**Review by poyenc — DISMISSED** [review]

**Review by poyenc — DISMISSED** [review]

File: example/ck_tile/01_fmha/codegen/ops/fmha_bwd.py [hunk 1]
```diff
@@ -271,7 +271,9 @@ def api(self) -> str:
                 per_hdim_case = per_hdim_case + FMHA_BWD_API_PER_HDIM_CASE.format(F_if=if_j, F_hdim=hdim, F_inner_dispatch=inners)
             if_i = 'if' if i == 0 else 'else if'
             per_dtypes = per_dtypes + FMHA_BWD_API_PER_DTYPE.format(F_if=if_i, F_dtype=dtype, F_hdim_case=per_hdim_case)
-
+        if not per_dtypes:
+            # empry string we add some ignore to supprise warning in api
```
Comment (aosewski) [inline example/ck_tile/01_fmha/codegen/ops/fmha_bwd.py]:
```suggestion
            # empty string we add some ignore to suppress warning in api
```

File: example/ck_tile/01_fmha/codegen/ops/fmha_fwd.py [hunk 2]
```diff
@@ -278,6 +278,9 @@ def api(self) -> str:
                 per_hdim_case = per_hdim_case + FMHA_FWD_API_PER_HDIM_CASE.format(F_if=if_j, F_hdim=hdim, F_inner_dispatch=inners)
             if_i = 'if' if i == 0 else 'else if'
             per_dtypes = per_dtypes + FMHA_FWD_API_PER_DTYPE.format(F_if=if_i, F_dtype=dtype, F_hdim_case=per_hdim_case)
+        if not per_dtypes:
+            # empry string we add some ignore to supprise warning in api
```
Comment (aosewski) [inline example/ck_tile/01_fmha/codegen/ops/fmha_fwd.py]:
```suggestion
            # empty string we add some ignore to suppress warning in api
```

File: example/ck_tile/01_fmha/codegen/ops/fmha_fwd_splitkv.py [hunk 3]
```diff
@@ -331,6 +331,9 @@ def api(self) -> str:
                 per_hdim_case = per_hdim_case + FMHA_FWD_API_PER_HDIM_CASE.format(F_if=if_j, F_hdim=hdim, F_inner_dispatch=inners)
             if_i = 'if' if i == 0 else 'else if'
             per_dtypes = per_dtypes + FMHA_FWD_API_PER_DTYPE.format(F_if=if_i, F_dtype=dtype, F_hdim_case=per_hdim_case)
+        if not per_dtypes:
+            # empry string we add some ignore to supprise warning in api
```
Comment (aosewski) [inline example/ck_tile/01_fmha/codegen/ops/fmha_fwd_splitkv.py]:
```suggestion
            # empty string we add some ignore to suppress warning in api
```

File: include/ck_tile/core/arch/arch.hpp [hunk 4]
```diff
@@ -82,14 +82,12 @@ CK_TILE_DEVICE void block_sync_lds_direct_load()
...
+CK_TILE_DEVICE void s_nop(index_t cnt = 0)
 {
 #if 1
-    asm volatile("\
-    s_nop 0 \n \
-    " ::);
+    asm volatile("s_nop %0" : : "n"(cnt) :);
 #else
-    __builtin_amdgcn_sched_barrier(0);
+    __builtin_amdgcn_sched_barrier(cnt);
```
Comment (aosewski) [inline include/ck_tile/core/arch/arch.hpp:90]: Will this have the same effect? https://llvm.org/docs/AMDGPUUsage.html#llvm-ir-intrinsics

File: example/ck_tile/01_fmha/codegen/ops/fmha_fwd.py (hunk 2 above)
Comment (carlushuang) [reply example/ck_tile/01_fmha/codegen/ops/fmha_fwd.py]: fixed

File: example/ck_tile/01_fmha/codegen/ops/fmha_fwd_splitkv.py (hunk 3 above)
Comment (carlushuang) [reply example/ck_tile/01_fmha/codegen/ops/fmha_fwd_splitkv.py]: fixed

File: include/ck_tile/core/arch/arch.hpp (hunk 4 above)
Comment (carlushuang) [reply include/ck_tile/core/arch/arch.hpp:90]: actually there is no intrinsic for s_nop, so usually will use inline asm version. If we want to use `__builtin_amdgcn_sched_barrier` usually just directly use this. So here you can ignore the #else

**Review by poyenc — APPROVED** [review]

**Review by illsilin — APPROVED** [review]

**Review by junliume — APPROVED** [review]

**Pull request merged**